src/request.rs

```diff
@@ -181,6 +182,37 @@ impl Request {
         self.do_call(Payload::Text(text, charset))
     }
 
+    /// Send a sequence of (key, value) pairs as form-urlencoded data.
+    ///
+    /// The `Content-Type` header is implicitly set to application/x-www-form-urlencoded.
+    /// The `Content-Length` header is implicitly set to the length of the serialized value.
+    ///
+    /// ```
+    /// #[macro_use]
+    /// extern crate ureq;
+    ///
+    /// fn main() {
+    /// let r = ureq::post("/my_page")
+    ///     .send_form(&[("foo", "bar"),("foo2", "bar2")]);
+    /// println!("{:?}", r);
+    /// }
+    /// ```
+    pub fn send_form<I, K, V>(&mut self, pairs: I) -> Response
+    where
+        I: IntoIterator,
+        I::Item: Borrow<(K, V)>,
+        K: AsRef<str>,
+        V: AsRef<str>,
```

Comment:
Nice thanks!

This breaks the pattern in the current API where we use `&str` and not `AsRef<str>`.

```rust
pub fn query(&mut self, param: &str, value: &str) -> &mut Request {
```

Maybe we should stick to the variant the rest of the API uses in this PR and then make a separate PR switching the entire API over to `AsRef<str>`? What do you think?

Reply:
Sure, sounds good! I changed the signature to be more consistent with the rest of the API.

```rust
pub fn send_form(&mut self, data: &[(&str, &str)]) -> Response {
```

Advancing to `AsRef` in a separate PR sounds very reasonable to me.